CONTRIBUTING.md

```diff
@@ -0,0 +1,227 @@
+
+      ```bash
+      $ make
+      ```
+
+-  When adding additional functionality, provide at least one
+   example script in the ``examples/`` folder. Have a look at other
+   examples for reference. Examples should demonstrate why the new
+   functionality is useful in practice and, if possible, compare it
+   to other methods available in pydicom.
+
+-  Documentation and high-coverage tests are necessary for enhancements to be
+   accepted. Bug-fixes or new features should be provided with 
+   [non-regression tests](https://en.wikipedia.org/wiki/Non-regression_testing).
+   These tests verify the correct behavior of the fix or feature. In this
+   manner, further modifications on the code base are granted to be consistent
+   with the desired behavior.
+   For the Bug-fixes case, at the time of the PR, this tests should fail for
+   the code base in master and pass for the PR code.
+
+
+-  At least one paragraph of narrative documentation with links to
+   references in the literature (with PDF links when possible) and
+   the example.
+
+-  The documentation should also include expected time and space
+   complexity of the algorithm and scalability, e.g. "this algorithm
+   can scale to a large number of samples > 100000, but does not
+   scale in dimensionality: n_features is expected to be lower than
+   100".
+
+You can also check for common programming errors with the following
+tools:
+
+-  Code with good unittest **coverage** (at least 80%), check with:
+
+  ```bash
+  $ pip install nose coverage
+  $ nosetests --with-coverage path/to/tests_for_package
+  ```
+
+-  No pyflakes warnings, check with:
+
+  ```bash
+  $ pip install pyflakes
+  $ pyflakes path/to/module.py
+  ```
+
+-  No PEP8 warnings, check with:
+
+  ```bash
+  $ pip install pep8
+  $ pep8 path/to/module.py
+  ```
+
+-  AutoPEP8 can help you fix some of the easy redundant errors:
+
+  ```bash
+  $ pip install autopep8
+  $ autopep8 path/to/pep8.py
+  ```
+
+Bonus points for contributions that include a performance analysis with
+a benchmark script and profiling output (please report on the mailing
+list or on the GitHub issue).
+
+Filing bugs
+-----------
+We use Github issues to track all bugs and feature requests; feel free to
+open an issue if you have found a bug or wish to see a feature implemented.
+
+It is recommended to check that your issue complies with the
+following rules before submitting:
+
+-  Verify that your issue is not being currently addressed by other
+   [issues](https://github.com/pydicom/pydicom/issues?q=)
+   or [pull requests](https://github.com/pydicom/pydicom/pulls?q=).
+
+-  Please ensure all code snippets and error messages are formatted in
+   appropriate code blocks.
+   See [Creating and highlighting code blocks](https://help.github.com/articles/creating-and-highlighting-code-blocks).
+
+-  Please include your operating system type and version number, as well
+   as your Python, pydicom, numpy, and scipy versions. This information
```

Comment:
scipy no, numpy yes if using Dataset.pixel_array. If using Dataset.pixel_array with compressed pixel data may also require the PIL/pillow/GDCM/jpeg_ls version numbers, depending on what's being used.